xformers/components/attention/csrc/cuda/mem_eff_attention/kernel_backward.h

```diff
@@ -83,9 +83,9 @@ struct AttentionBackwardKernel {
     accum_t* delta_ptr; // [Mq, nH]
 
     // Output tensors
-    scalar_t* grad_query_ptr; //  [Mq, nH, K]
-    scalar_t* grad_key_ptr; //    [Mk, nH, K]
-    scalar_t* grad_value_ptr; //  [Mk, nH, Kv]
+    output_t* grad_query_ptr; //  [Mq, nH, K]
+    output_t* grad_key_ptr; //    [Mk, nH, K]
+    output_t* grad_value_ptr; //  [Mk, nH, Kv]
```

Comment:
nit: I think this can be a bit misleading.

Indeed, if we define the `scalar_t` as the dtype of `query` / `key` / value`, and `output_t` the `dtype` of the output of the attention, then the gradients of `query` / `key` / `value` should have the same `dtype` as their respective `query` / `key` / `value`, and not on the output dtype.

If ever we make in the forward that `scalar_t != output_t`, then this could be an issue.